### PR TITLE
[SPARK-39789] Remove unused method and redundant throw exception declare

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
Review code and found some issue about unused method and redundant throw exception declare.

### Why are the changes needed?
Eliminate redundant information and make the code cleaner

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

File: launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java [hunk 1]
```diff
@@ -208,7 +208,7 @@ public void testExamplesRunnerNoMainClass() throws Exception {
   }
 
   @Test
-  public void testExamplesRunnerWithMasterNoMainClass() throws Exception {
```
Comment (srowen) [inline launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java:211]: These are `@Test` methods, so the overly broad signature doesn't matter (they are not called by code)

**Review by srowen — COMMENTED** [review]
I think this sort of thing is just too minor to bother with

File: launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java (hunk 1 above)
Comment (panbingkun) [reply launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java:211]: AbstractCommandBuilder.buildClassPath is not test methods.
In fact, When I review launcher module, I found the following **two kinds** of problems: https://github.com/apache/spark/pull/37199 & https://github.com/apache/spark/pull/37200

#### Should I close the above two PR?

Comment (srowen) [reply launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java:211]: THey aren't bad changes, I'm neutral; just generally if there's no real functional improvement, it's not worth opening and reviewing it

**Comment by panbingkun** [comment]
i will close

**Pull request closed without merging**